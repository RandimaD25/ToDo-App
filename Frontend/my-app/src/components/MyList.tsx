import { Button } from 'react-bootstrap'

interface ItemProps {
    todo: TodoType;
    onRemoveTodo: (todoID:number) => void;
    onDoneTodo: (todoId: number) => void;
  }

const MyList: React.FC<ItemProps> = ({todo, onRemoveTodo, onDoneTodo}) => {

    const onDelete = () => {
        onRemoveTodo(todo.id);
    }

    const onDone = () => {
        console.log("done"); 
        onDoneTodo(todo.id);
    }

    return (
        <ul data-testid="todo-1" id='list'>
            <li
                className='align-items-left' 
                style={{marginInline: "15rem"}}>
                <div className='d-flex mt-4 gap-4' style={{display:"flex", justifyContent: "space-between"}}>
                {todo.description}
                    <div className='d-flex gap-4 align-items-right'>
                        <Button onClick={onDone} className='btn btn-success' hidden={todo.flag===true}>Done</Button>
                    
                        <Button onClick={onDelete} className='btn btn-danger'>Delete</Button>
                    </div>
                </div>  
            </li>  
        </ul>    
    )
}

export default MyList;

