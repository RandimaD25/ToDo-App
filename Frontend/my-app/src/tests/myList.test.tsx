import { render, fireEvent } from "@testing-library/react";
import MyList from "../components/myList";
import renderer from "react-test-renderer";

describe("MyList component", () => {
  test("render the MyList component", () => {
    const mockOnRemoveTodo = jest.fn();
    const mockOnDoneTodo = jest.fn();

    const todo = {
      id: 1,
      description: "Test Todo",
      flag: false,
    };

    const { getByText } = render(
      <MyList
        todo={todo}
        onRemoveTodo={mockOnRemoveTodo}
        onDoneTodo={mockOnDoneTodo}
      />
    );

    fireEvent.click(getByText("Done"));
    expect(mockOnDoneTodo).toHaveBeenCalledWith(todo.id);

    fireEvent.click(getByText("Delete"));
    expect(mockOnRemoveTodo).toHaveBeenCalledWith(todo.id);
  });  

  test("should call onDoneTodo when the done button is clicked", () => {
    const todo = { id: 1, description: 'Test todo', flag: false };
    const onRemoveTodo = jest.fn();
    const onDoneTodo = jest.fn();

    const { getByText } = render(
      <MyList todo={todo} onRemoveTodo={onRemoveTodo} onDoneTodo={onDoneTodo} />
    );
    const doneButton = getByText('Done');
    fireEvent.click(doneButton);

    expect(onDoneTodo).toHaveBeenCalledWith(todo.id);
  });

  test("calls onRemoveTodo when the delete button is clicked", () => {
    const mockTodo = { id:1, description: 'Test todo', flag: false };
    const mockOnRemoveTodo = jest.fn();
    const mockOnDoneTodo = jest.fn();

    const { getByText } = render(
      <MyList
        todo={mockTodo}
        onRemoveTodo={mockOnRemoveTodo}
        onDoneTodo={mockOnDoneTodo}
      />
    );
    const deleteButton = getByText('Delete');
    fireEvent.click(deleteButton);

    expect(mockOnRemoveTodo).toHaveBeenCalledWith(mockTodo.id);
  })
});

describe("MyList component Snapshot", ()=> {
  test("should matches DOM Snapshot", () => {
    const todo = {
      id: 1,
      description: "Test Todo",
      flag: false,
    };

    const onRemoveTodo = jest.fn();
    const onDoneTodo = jest.fn();

    const tree = renderer
      .create(
        <MyList
          todo={todo}
          onRemoveTodo={onRemoveTodo}
          onDoneTodo={onDoneTodo}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
})
