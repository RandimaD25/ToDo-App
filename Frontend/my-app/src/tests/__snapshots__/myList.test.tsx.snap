// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList component Snapshot should matches DOM Snapshot 1`] = `
<ul
  data-testid="todo-1"
  id="list"
>
  <li
    className="align-items-left"
    style={
      {
        "marginInline": "15rem",
      }
    }
  >
    <div
      className="d-flex mt-4 gap-4"
      style={
        {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      Test Todo
      <div
        className="d-flex gap-4 align-items-right"
      >
        <button
          className="btn btn-success btn btn-primary"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          type="button"
        >
          Done
        </button>
        <button
          className="btn btn-danger btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </li>
</ul>
`;
